packages/node_modules/@webex/plugin-meetings/src/reachability/index.js

```diff
@@ -0,0 +1,283 @@
+    this.gatherReachability();
+  }
+
+  setup = () => {
+    this.processedClusterResults = [];
+    this.peerConnectionMap = {};
+  }
+
+  /**
+   * fetches reachability data
+   * @returns {object} reachability data
+   */
+  gatherReachability = () => {
+    this.setup();
+    // Reset reachability result to empty to ensure no stale data
+    try {
+      window.localStorage.setItem(REACHABILITY.localStorage, JSON.stringify({}));
+
+      // Get clusters from Orpheus
+      return this.reachabilityRequest.getClusters()
+        .then((clusters) =>
+          // Perform Reachability Check
+          this.performReachabilityCheck(clusters)
+            .then((response) => {
+              window.localStorage.setItem(REACHABILITY.localStorage, JSON.stringify(response));
+              return response;
+            }))
+        .catch((error) => {
+          console.info(`Error in calling getClusters() :${error}`);
+          return {};
+        });
+    }
+    catch (error) {
+      console.log('There was an error while accessing LocalStorage', error);
+      return {};
+    }
+  };
+
+  /**
+   * fetches reachability data
+   * @param {object} clusterList
+   * @returns {object} reachability check results
+   */
+  performReachabilityCheck = (clusterList) => {
+    if (!clusterList || !clusterList.length) {
+      return Promise.resolve({});
+    }
+
+    let localSDPList = [];
+
+    return new Promise(((resolve) => {
+      this.getLocalSDPForClusters(clusterList)
+        .then((localSDPData) => {
+          if (!localSDPData || !Object.keys(localSDPData).length || !localSDPData.localSDPList || !localSDPData.peerConnection) {
+            // TODO: handle the error condition properly and try retry
+            console.info('Local SDP is empty or has missing elements..returning.');
+            resolve({});
+          }
+
+          this.peerConnectionMap = localSDPData.peerConnection;
+          ({localSDPList} = localSDPData);
+          this.setRemoteOfferForEachCluster(localSDPList)
+            .then((response) => {
+              resolve(response);
+            });
+        })
+        .catch((error) => {
+          console.error(`Error in getLocalSDPForClusters :${error}`);
+          resolve({});
+        });
+    }));
+  };
+
+  /**
+   * sets remote offer for each clusters
+   * @param {object} localSDPList local SDP generated
+   * @returns {object} reachability check results
+   */
+  setRemoteOfferForEachCluster = (localSDPList) => {
+    const iceResult = {};
+    return new Promise((resolve) => {
+      this.reachabilityRequest.remoteSDPForClusters(localSDPList)
+        .then((remoteSDPResponse) => {
+          const remoteSDPList = remoteSDPResponse.answers;
+          if (!remoteSDPList || !remoteSDPList.length) {
+            console.info('Remote SDP is empty ');
+            this.cleanUpPeerConnections(this.peerConnectionMap);
+            resolve({});
+          }
+          remoteSDPList.forEach((remoteSdp) => {
```

Comment:
Are there more than one SDP here? You call resolve below while iterating through the array. If there are more than one it will potentially resolve/reject before it completes.

Reply:
There is multiple sdps and we only have to resolve when all gets resolved so that we can make a network request